pkg/compiler/compiler_test.go

```diff
@@ -0,0 +1,107 @@
+package compiler
+
+import (
+	"testing"
+
+	"evylang.dev/evy/pkg/assert"
+	"evylang.dev/evy/pkg/code"
+	"evylang.dev/evy/pkg/evaluator"
+	"evylang.dev/evy/pkg/parser"
+)
+
+type compilerTestCase struct {
+	input                string
+	expectedConstants    []interface{}
+	expectedInstructions []code.Instructions
+}
+
+func TestGlobalVarStatements(t *testing.T) {
+	tests := []compilerTestCase{
+		{
+			input:             "x := 1\nx = x",
+			expectedConstants: []interface{}{1},
+			expectedInstructions: []code.Instructions{
+				code.Make(code.OpConstant, 0),
+				code.Make(code.OpSetGlobal, 0),
+				code.Make(code.OpGetGlobal, 0),
+				code.Make(code.OpSetGlobal, 1),
+			},
+		},
+		{
+			input:             "x := 2\nx = x",
+			expectedConstants: []interface{}{2},
+			expectedInstructions: []code.Instructions{
+				code.Make(code.OpConstant, 0),
+				code.Make(code.OpSetGlobal, 0),
+				code.Make(code.OpGetGlobal, 0),
+				code.Make(code.OpSetGlobal, 1),
+			},
+		},
+	}
+	runCompilerTests(t, tests)
```

Comment:
maybe fold `runCompilerTests` in here.
some people pull out the testCases as global vars, especially when used in different test functions.